server/src/main/java/org/opensearch/action/admin/cluster/node/stats/NodeStats.java

```diff
@@ -374,6 +389,9 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getVersion().onOrAfter(Version.V_2_6_0)) {
             out.writeOptionalWriteable(weightedRoutingStats);
         }
+        if (out.getVersion().onOrAfter(Version.V_2_6_0)) {
```

Comment:
BWC mixedcluster test fails as this change is not in 2.x

Reply:
You'll want to make this `Version.V_3_0_0` in this PR. Then do a follow up to change it after the backport. This process is not ideal but we definitely want a passing gradle check on all PRs before they are merged, and as you said this will not pass the tests as it is.

Reply:
Just to be clear, that follow up PR will also fail until both 2.x and main are in sync, right?

Reply:
The backport will cause the BWC tests on main to break, but both the backport and follow up PRs will pass their tests. Only unrelated PRs that come in between the backport and follow up PR will see failing tests.

Reply:
Got it, let me change it